streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -361,6 +363,25 @@ class StreamingContext private[streaming] (
 
   /**
    * Create a input stream that monitors a Hadoop-compatible filesystem
+   * for new files and reads them using the given key-value types and input format.
+   * Files must be written to the monitored directory by "moving" them from another
+   * location within the same file system. File names starting with . are ignored.
+   * @param directory HDFS directory to monitor for new file
+   * @param conf Hadoop configuration
+   * @tparam K Key type for reading HDFS file
+   * @tparam V Value type for reading HDFS file
+   * @tparam F Input format for reading HDFS file
+   */
+  def fileStream[
```

Comment:
Why the wiring to binaryRecords? Those two parameters are only relevant to
DStream, so nothing to pass on down to the RDDs
On Feb 2, 2015 7:30 PM, "Jeremy Freeman" <notifications@github.com> wrote:

> In
> streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala
> <https://github.com/apache/spark/pull/3803#discussion_r23980541>:
>
> > @@ -361,6 +363,25 @@ class StreamingContext private[streaming] (
> >
> >    /**
> >     * Create a input stream that monitors a Hadoop-compatible filesystem
> > +   * for new files and reads them using the given key-value types and input format.
> > +   * Files must be written to the monitored directory by "moving" them from another
> > +   * location within the same file system. File names starting with . are ignored.
> > +   * @param directory HDFS directory to monitor for new file
> > +   * @param conf Hadoop configuration
> > +   * @tparam K Key type for reading HDFS file
> > +   * @tparam V Value type for reading HDFS file
> > +   * @tparam F Input format for reading HDFS file
> > +   */
> > +  def fileStream[
>
> Fine with me, though would require wiring the extra arguments (filter and
> newFilesOnly) through to all the binary record methods (because they'll
> also need the conf). Should we expose those arguments in binaryRecords?
> Or just use their defaults?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/3803/files#r23980541>.
>